presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -293,6 +302,16 @@ private boolean isLastKilledQueryGone()
         if (lastKilledQuery == null) {
             return true;
         }
+
+        // If the lastKilledQuery is marked as leaked by the ClusterMemoryLeakDetector we consider the lastKilledQuery as gone,
+        // so that the ClusterMemoryManager can continue to make progress even if there are leaks.
+        // Even if the weak references to the leaked queries are GCed in the ClusterMemoryLeakDetector, it will mark the same queries
+        // as leaked in its next run, and eventually the ClusterMemoryManager will make progress.
```

Comment:
`ClusterMemoryManager` can make progress, but will the leaked queries' reservations be ultimately freed?